include/espi.h

```diff
@@ -0,0 +1,446 @@
+
+/*
+ * Copyright (c) 2019 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for eSPI driver
+ */
+
+#ifndef ZEPHYR_INCLUDE_ESPI_H_
+#define ZEPHYR_INCLUDE_ESPI_H_
+
+#include <misc/__assert.h>
+#include <misc/slist.h>
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief ESPI Driver APIs
+ * @defgroup espi_interface ESPI Driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/*
+ * eSPI VW channel packets contain 4 VWire states.
+ */
+#define ESPI_VW_SRC_ID0         0
+#define ESPI_VW_SRC_ID1         1
+#define ESPI_VW_SRC_ID2         2
+#define ESPI_VW_SRC_ID3         3
+#define ESPI_VW_SRC_ID_MAX      4
+
+
+/** @cond INTERNAL_HIDDEN */
+struct espi_vwire {
+	u8_t	idx;
+	u8_t	bit;
+	u8_t	dir;
+};
+
+/**
+ * @endcond
+ */
+
+/**
+ * @brief ESPI I/O mode capabilities
+ */
+enum espi_io_mode {
+	DUAL_LINES = BIT(0),
+	QUAD_LINES = BIT(1),
+	SINGLE_LINE = BIT(2),
+};
+
+/**
+ * @brief ESPI frequency capabilities
+ */
+enum espi_freq {
+	MAX_FREQ_20MHz,
+	MAX_FREQ_25MHz,
+	MAX_FREQ_33MHz,
+	MAX_FREQ_50MHz,
+	MAX_FREQ_66MHz
+};
+
+/**
+ * @brief ESPI channels support
+ */
+
+enum espi_channel {
+	PERIPHERAL_CHANNEL	= BIT(0),
```

Comment:
Add ESPI_CHANEL_ as prefix to all values here (remove _CHANNEL from existing names obviously)